### PR TITLE
add docs for `check_and_build_global_config()`

### LINKED ISSUE #268: fix typo, remove unncessary functions, and add example for `update_config_workflow`
Thanks for the excellent review @yucongalicechen .  Please could you make a PR to fix those things?  Let's also double check there are no private functions that are not use, or public functions that we want to make private.

If @bobleesj finds anything else we can put that into the same PR.

I also wanted to add into the docs what happens in the update config workflow when it is triggered, so reproduce what the user would see and give an example of how they would respond.  It would be great for me if that could be in the same PR?

_Originally posted by @sbillinge in https://github.com/diffpy/diffpy.utils/issues/264#issuecomment-2558498441_

### PR DESCRIPTION
**Pull request by yucongalicechen**
closes #268
@sbillinge ready for review

### CODE REVIEW

File: doc/source/examples/tools_example.rst [hunk 1]
```diff
@@ -95,12 +96,32 @@ it will only run once.  However, if you want to bypass this behavior,
 ``check_and_build_global_config()`` takes an optional boolean ``skip_config_creation`` parameter that
 could be set to ``True`` at runtime to override the config creation.
 
+What happens when you run ``check_and_build_global_config()``?
```
Comment (yucongalicechen) [inline doc/source/examples/tools_example.rst:99]: add example

File: doc/source/examples/tools_example.rst [hunk 2]
```diff
@@ -111,7 +132,7 @@ Locate the file ``diffpyconfig.json``, in your home directory and open it in an
         "owner_orcid": "0000-0000-4321-1234"
     }
 
-   Then you can edit the username and email as needed, make sure to save your edits.
+Then you can edit the username and email as needed, make sure to save your edits.
```
Comment (yucongalicechen) [inline doc/source/examples/tools_example.rst:135]: reformat

File: doc/source/utilities/tools_utility.rst [hunk 3]
```diff
@@ -5,12 +5,17 @@ Tools Utility
...
 The ``diffpy.utils.tools`` module provides tool functions for use with diffpy apps.
 
-- ``get_user_info()``: This function is designed for managing and tracking username and email information.
-
+- ``get_user_info()``: This function is designed for managing and tracking user information (name, email, orcid).
   Developers can use this function to simplify the process of loading, merging, and saving information consistently and easily.
   Additionally, it saves the effort of re-entering information, and allows overriding current information by
   passing parameters.
 
+- ``check_and_build_global_config()``: This function helps create a global configuration file
```
Comment (yucongalicechen) [inline doc/source/utilities/tools_utility.rst:13]: add a short description

File: src/diffpy/utils/tools.py [hunk 4]
```diff
@@ -4,29 +4,7 @@
...
-    """
-    obj = obj if obj is not None else {}
-    for key, value in copy(obj).items():
-        if not value:
-            del obj[key]
-    return obj
-
-
-def stringify(obj):
+def _stringify(obj):
```
Comment (yucongalicechen) [inline src/diffpy/utils/tools.py:7]: make this function private and remove unncessary functions

**Review by sbillinge — COMMENTED** [review]
beautifully done. Just a couple of suggestions in line

File: doc/source/examples/tools_example.rst [hunk 5]
```diff
@@ -110,10 +110,8 @@ An example of the prompts you may see is:
     Please enter your orcid ID if you know it: 0000-0000-0000-0000
 
 
-After receiving the inputs, the function will write the following to the file:
-
-.. code-block:: python
-    {"owner_name": "Jane Doe", "owner_email": "janedoe@email.com", "owner_orcid": "0000-0000-0000-0000"}
+After receiving the inputs, the function will write the information to
+the `diffpyconfig.json` file in your home directory.
```
Comment (yucongalicechen) [inline doc/source/examples/tools_example.rst:114]: I deleted the example here - there are examples both before and after this section

File: doc/source/utilities/tools_utility.rst [hunk 6]
```diff
@@ -11,8 +11,8 @@ The ``diffpy.utils.tools`` module provides tool functions for use with diffpy ap
   passing parameters.
 
 - ``check_and_build_global_config()``: This function helps create a global configuration file
-  that can be used by ``get_user_info()``.
-  If no existing configuration file is found, and the user allows inputs, this function prompts for information.
+  that can be used by, for example, ``get_user_info()``.
+  If no existing configuration file is found, this function prompts for information.
```
Comment (yucongalicechen) [inline doc/source/utilities/tools_utility.rst:15]: deleted "allows inputs"

**Comment by yucongalicechen** [comment]
@sbillinge ready for review!

**Pull request merged**